### ISSUE TITLE
Make it possible to import packages on container with lower than 2 GB of memory

### ISSUE THREAD

**Issue by schlichtanders** [JuliaLang/julia#46912]
This is actually a really crucial problem I am facing currently. I already pushed it at the specific package which I was not able to import, [SciMLSensitivity](https://github.com/SciML/SciMLSensitivity.jl/issues/734#issuecomment-1257070533), but @ChrisRackauckas mentioned this is rather a generic Julia issue.

What happens? julia is killed as soon as I merely import a package...
![image](https://user-images.githubusercontent.com/4255480/192276012-eb7828f0-0c0b-44d6-b8cc-3eea8632c8a9.png)

This happens, because the docker container is restricted to 2 GB of RAM, while loading this package needs more than 2 GB of RAM (or at least in this default setup without any additional configuration it needs more).

Hence I am unable to use binder for SciML, and also anyone else is unable to use mybinder for SciML. A real pity.
It would be great if loading a package does not kill a moderate machine like a binder instance with 2 GB of RAM.

**Comment by gbaraldi** [comment]
1.8.2 might make this a bit better because it makes julia more aware of memory constraints. It should release in a couple days if no issues are found

**Comment by PallHaraldsson** [comment]
New in 1.9, can you try something like:
````
julia --heap-size-hint=2G
```

I think compilation of packages makes new julia processes, then they have their own memory and GC, and the limitation is likely not inherited (but should it?). You can also try limiting parallel compilation. Maybe just disable it with:
```
ENV["JULIA_NUM_PRECOMPILE_TASKS"] = 1
```

**Comment by maleadt** [comment]
> New in 1.9, can you try something like:
> 
> ```
> julia --heap-size-hint=2G
> ```

No need for that, it will be auto-detected. Only use `--heap-size` if you want (or need) to restrict further.

**Comment by schlichtanders** [comment]
It would also be great to have a way to change julia's import behaviour from within the running julia kernel

**Comment by schlichtanders** [comment]
sorry, wrong button

**Comment by PallHaraldsson** [comment]
This might be fixed already with: https://github.com/JuliaLang/julia/pull/46796

I would check with master/nightly at the download page. Hopefully backported to 1.8.2 which should appear soon (it was merged recently so might not have made it, if that's not fixed then 1.8.3, that I suppose might come, though do not see planned yet).

**Comment by maleadt** [comment]
1.8.2 is imminent, and should fix this.

**Comment by schlichtanders** [comment]
This is unfortunately not fixed in julia 1.8.2

the kernel still dies when importing
![image](https://user-images.githubusercontent.com/4255480/194777164-c1d28dad-46ae-4528-a50e-4c6e3e93f198.png)

**Comment by maleadt** [comment]
What distro are you trying this on? Does it use cgroupv1 or v2? https://github.com/JuliaLang/julia/pull/46796#issuecomment-1272117544 may be relevant.
